packages/terra-search-field/src/SearchField.jsx

```diff
@@ -70,6 +70,11 @@ const propTypes = {
    * The value of search field.  Use this to create a controlled search field.
    */
   value: PropTypes.string,
+
+  /**
+   * Callback ref to pass into the inner input component.
+   */
+  inputFieldRefCallback: PropTypes.func,
```

Comment:
Depending on the needs of your team, another possible solution could be a first class autoFocus property that is spread onto the input. The refCallback is a wildcard for doing anything custom, but the autoFocus approach has an explicit API. 

```
<SearchField autoFocus />
```
VS
```
class Example extends React.Component {
  constructor() {
    super();
  }

  componentDidMount() {
    this.input.focus();
  }

  render() {
    return <SearchFieldExampleTemplate inputFieldRefCallback={(input) => { this.input = input; }} />;
  }
}
```

There are tradeoffs for each approach. The AutoFocus will only work when the component mounts but has an explicit API. The refCallback is more extensible but loosely couples functionality and allows many variations of behavior within an application.

A request has recently been logged to autoFocus the select component and if possible it would be nice to settle on a pattern going forward or if each component's ability to focus is case by case.

Reply:
My team hasn't finalized the requirements for our search feature yet, but we may want to focus it in more scenarios other than just when the component mounts, so it is nice to have the ability to implement the focusing ourselves.

Also, I initially had my example component on the terra-dev-site performing the focus on componentDidMount, but that causes the search-field site to automatically focus on that one example instead of loading the top of the page. I'm afraid an autofocus example would cause the same behavior.

Reply:
Yeah I'm fine with changing it to `inputRefCallback`! Will make the change.